### PR TITLE
fix VERSIONS

### PR DESCRIPTION
**Pull request by JelleZijlstra**
See #5328. I mostly checked manually for things that imported in 2.7, and checked the Python 3 documentation for some of the remaining ones.

I think a few of the underscored modules may also have existed in earlier versions, but that's harder to figure out since they're undocumented.

### CODE REVIEW

**Comment by JelleZijlstra** [comment]
I feel like there's something wrong in pytype's version handling. For this one:
```
stubs/tornado/tornado/testing.pyi (2.7): pytype.load_pytd.BadDependencyError: Can't find pyi for '_collections_abc', referenced from 'collections.abc'
```
I don't see any references to `collections.abc` in the `tornado.testing` stub. pytype should be using only `stdlib/@python2/collections.pyi`, which doesn't have references to `collections.abc`.

**Comment by srittau** [comment]
I am not 100% convinced that we should change this. Our stubs don't really support version prior to 3.6. That said, I don't really mind. But you also changed stubs that have a separate version in the `@python2` directory as starting with `2.7`. I don't think this is how this file is supposed to work. From what I understand, it should only list the versions supported version by the main `stdlib` directory.

**Comment by JelleZijlstra** [comment]
> From what I understand, it should only list the versions supported version by the main stdlib directory.

Our documentation says "`VERSIONS` file lists the oldest *supported* Python version where the module is available." I guess we can take out the word "supported" if we merge this PR, but it's still true that these modules are available in 2.7. However, type checkers should look in `@python2` first.

cc @rchen152 for pytype

**Comment by srittau** [comment]
FWIW, I'd think it makes sense to list them as 2.7 here, but we have to double check that this will work with both mypy and pytype.

**Review by srittau — APPROVED** [review]
Approving if the Python 2.7 change works with mypy and pytype.

**Comment by rchen152** [comment]
pytype is failing here because this is a change to the semantics of `VERSIONS`; previously, it listed the lowest supported version for the stub in `stdlib/foo`, ignoring `stdlib/@python2/foo`.

I don't think this is hard to change in pytype - I should be able to release a new version with this change in a couple days if nothing unexpected comes up.

**Comment by JelleZijlstra** [comment]
Thanks! I'll take a look tonight to see if I can provide a fix for pytype.

**Comment by srittau** [comment]
@JelleZijlstra See als google/pytype#903, as that may conflict.

File: stdlib/VERSIONS [hunk 1]
```diff
@@ -14,121 +14,121 @@ _dummy_threading: 2.7
...
 _py_abc: 3.7
 _pydecimal: 3.6
 _random: 2.7
 _sitebuiltins: 3.6
 _stat: 3.6
-_thread: 3.6
+_thread: 2.7
 _threading_local: 3.6
-_tkinter: 3.6
+_tkinter: 3.0
```
Comment (Akuli) [inline stdlib/VERSIONS:30]: Python 2 has `_tkinter` and `Tkinter` (but there are no stubs for `Tkinter`). Python 3 has `_tkinter` and `tkinter`.

```suggestion
_tkinter: 2.7
```

**Review by Akuli — CHANGES_REQUESTED** [review]

**Comment by rchen152** [comment]
Just realized I forgot to close the loop on this - pytype 2021.05.04 and later contains both PRs to fix pytype's handling of VERSIONS (https://github.com/google/pytype/pull/903, https://github.com/google/pytype/pull/906). (The PR history looks a little weird because we import and close PRs rather than merging them, but the changes are indeed merged.)

**Comment by JelleZijlstra** [comment]
Thanks! Since @Akuli's change was also addressed I'll merge this after we've upgraded pytype and pyright in our CI. I'll do that tonight unless someone else gets to it first.

**Pull request merged**